common/os.cpp

```diff
@@ -287,6 +287,11 @@ Some auxillary functionalities might be affected. Please report this message if
                 res = str;
                 res += "\\";
             }
+            else if ( f == user_documents)
+            {
+                std::string msg = "Failed to get Documents directory";
```

Comment:
Pls parse and print the value of `HRESULT hr = SHGetKnownFolderPath(...` - there are additional failure cases as well